src/GraphQL/Resolvers/NameFieldResolver.cs

```diff
@@ -20,15 +20,7 @@ public static object Resolve(object source, string name)
                 return null;
             }
 
-            var prop = source.GetType()
-                .GetProperty(name, _flags);
-
-            if (prop == null)
-            {
-                throw new InvalidOperationException($"Expected to find property {name} on {source.GetType().Name} but it does not exist.");
-            }
-
-            return prop.GetValue(source, null);
+            return source.GetPropertyValue(name);
```

Comment:
I think this is the only change I'm mildly concerned about.  Mind moving the exception to the `GetPropertyValue` method?  I think that provides helpful debugging - perhaps the error should only be thrown in debug mode?  Maybe that would cause too much confusion with different behavior?  It appears that method wasn't being used by anything else.